### ISSUE TITLE
Visitors Exit Map Immediately

### ISSUE THREAD

**Issue by crazyjat** [OrionFive/Hospitality#153]
I play with only this mod enabled and every time, either after a certain amount of time has passed or I have built a colony of a certain size (not sure which) guests will continue to show up, but upon doing so will exit the map immediately. 

It doesn't matter if the area is safe or not, after a certain point, every time visitors show up, they exit the map immediatly.

I love this mod, but kinda pointless after that starts happening.

Current colony screenshot:
https://imgur.com/35CynX0 

Log:
https://git.io/vpAHt

**Comment by pl-wiltschek** [comment]
I can't see what could go wrong from the log. Did the described incident happen within the session of the log?

**Comment by crazyjat** [comment]
I have looked at the logs every time this happens and they aren't helpful. What I can tell you is this happens every time after a certain point with every new game. I have tried uninstalling and reinstalling the game over and over again. I can only assume everyone is experiencing this issue after a certain point in their game. 

I have a new game with a bunch of mods activated and it just happened again. The log is located at https://git.io/vhcmK and provides no useful information about what happened. 

Is there some way to enable additional logging for your mod that I should enable?

**Comment by OrionFive** [comment]
Hey, sorry I didn't have time to look into your issue yet. RimWorld 1.0 is prereleased so I'll focus my efforts on the update. Afterwards all old bugs will be replaced with new ones anyway.

**Comment by OrionFive** [comment]
Reopen if still present in 1.0.

**Comment by crazyjat** [comment]
Issue still occurs in 1.0. 

Issue occurs when more than 10 colonists are present in a colony.

**Comment by OrionFive** [comment]
Not with less?

**Comment by crazyjat** [comment]
With the previous version at least, with 11 colonists, visitors would always exit the map immediately, and by that I mean that they enter the map, and walk clear across to another side of the map and exit without ever stopping. My current game I have 15 colonists and visitors exit immediately. No visitor will ever stop, just exit. 

I did not notice this time around if it happened exactly at 11, but I can send some of my colonists to settle another colony so I have only 10 or fewer colonists in each settlement. In the previous version at least, this would cause visitors to again visit my colonies and actually stay.

**Comment by OrionFive** [comment]
I haven't been able to reproduce this yet. Until I do, I won't be able to do anything about it... So any information you can give me to recreate the situation will help.

**Comment by crazyjat** [comment]
It is really easy to repro. Just start a new game with only hugslib and this mod, go into debug menu and add colonists until you have 11, then when the next visitor arrives, notice they leave immediately without stopping at the colony.

**Comment by OrionFive** [comment]
Thanks, I'll try that.

**Comment by OrionFive** [comment]
I couldn't reproduce it with my own game, but I could find an issue using your old savegame from B18.

Guests have a point that they travel to when they arrive. It turns out that when they can't reach it, they will leave again. In your savegame, that point was exactly in your "chimney" behind the coolers, where no one can get. So they left again, every time.

I'll see if I can write some code for them to pick a smarter point.

**Comment by OrionFive** [comment]
Done.

**Comment by MonochromeMagpie** [comment]
I'm experiencing this now...

**Comment by OrionFive** [comment]
This usually means your guests can't reach their designated zone or you an incompatible mod.
Try and set the guest zone to Home, make it the default and wait for the next group.

**Comment by khamseen** [comment]
I've just had this issue happen to me too. I thought it was strange I hadn't seen any guests in my guest room for a while and when some more showed up just now I watched them closely and saw them just walk straight across the map and out again.
Is there a way to visualise where they are meant to try and head when they first arrive so I can find out where I need to make accessible? 
I had them set to Home zone already and there are doors in my outer wall so they should have been able to get to that zone without an issue but I'm guessing it's a specific set point rather than just the zone itself. For now I've set them to unrestricted to see if that affects the next visitors. But without knowing where they're trying to head to, it's pretty impossible to fix this.

**Comment by khamseen** [comment]
Setting them to unrestricted did not help. Next group just walked right past my base too.

**Comment by OrionFive** [comment]
Well there are different causes for this. One would be that your guest zone includes a very specific spot that is walled in and unreachable. Obviously you fix it by removing the spot from the zone. 
Another is a mod that adds new navigation restrictions. For example most door / lock mods. It often doesn't matter if the doors are locked or not, they will break visitor pathfinding when they arrive. 

The next one is errors occurring when the group arrives but has fully spawned. They will cancel their visit and leave again.

**Comment by khamseen** [comment]
I do run the door lock mod, but I've had that active in my last three playthroughs and it hasn't caused a problem with guests, the only doors that I use it on are my prison to keep guests out and prisoners in. Normal caravans from the same factions as the guests can still enter by base and make their way to my trading spot right in the middle of my base, next to the guest quarters so I have no idea why the guests can't. 

I have been checking the debug log as guests arrive and haven't seen any errors reported. Is there a debug option to spawn a group of visitors using standard settings so I could load a copy of my world at least and try spamming them to see which areas I need to adjust? Without knowing what exact point they have picked to 'arrive' at, it's not an easy task knowing what needs to change. 

Short of ripping out all of my walls throughout the base and seeing if that helps, I'm kind of at a loss and that bugs me because I'm all about the guests.

**Comment by OrionFive** [comment]
Caravans don't use the same algorithm for navigation... or anything else actually.

Is the prison included in the area for the guests? Then it could be that by chance they now want to go there which they didn't in an earlier playthrough. 

I think by enabling "drawing lords" you can see where they want to go. Problem is since they can't get there they immediately go somewhere else, so you probably can't see it.

So just make sure the guest area doesn't include parts that are not reachable.

**Comment by khamseen** [comment]
I think you were right, they'd picked an area inside my prison to head for first which they couldn't get to. I set them back to home zone and temporarily removed the prison from my home zone and the next group who arrived happily entered the base and are now hanging out again!

It was just something I hadn't considered because I've used these mods together multiple times and always lock my prison and leave guests on home zone without any problems but apparently this time they got fixated on it. Maybe the lore of the bug infestation which decimated my prison population was appealing haha. 

Now that I've got them working again I'll make them their own zone which covers everywhere except my prison and hopefully that'll be good enough for them. 
Sorry to have bugged you on this and dragged up a closed thread. I just couldn't live without my visitors. :)
